tests/tests_data/streaming/test_dataset.py

```diff
@@ -0,0 +1,201 @@
+    assert len(dataset_iter) == 41
+    process_1_1 = list(dataset_iter)
+    assert len(process_1_1) == 41
+    dataset_iter = iter(dataset)
+    assert len(dataset_iter) == 50
+    process_1_2 = list(dataset_iter)
+    assert len(process_1_2) == 50
+
+    dataset = StreamingDataset(name="choco", cache_dir=tmpdir)
+    dataset.distributed_env = _DistributedEnv(2, 1)
+    assert len(dataset) == 41
+    dataset_iter = iter(dataset)
+    process_2_1 = list(dataset_iter)
+    assert len(process_2_1) == 41
+    dataset_iter = iter(dataset)
+    assert len(dataset_iter) == 50
+    process_2_2 = list(dataset_iter)
+    assert len(process_2_2) == 50
+
+    assert len([i for i in process_1_1 if i in process_2_1]) == 0
+    assert len([i for i in process_1_2 if i in process_2_2]) == 0
+
+
+def test_streaming_dataset_distributed_no_shuffle(tmpdir):
+    seed_everything(42)
+
+    cache = Cache(tmpdir, chunk_size=10)
+    for i in range(101):
+        cache[i] = i
+
+    cache.done()
+    cache.merge()
+
+    dataset = StreamingDataset(name="choco", cache_dir=tmpdir, shuffle=False)
+
+    assert isinstance(dataset.shuffle, NoShuffle)
+
+    for i in range(101):
+        assert dataset[i] == i
+
+    dataset.distributed_env = _DistributedEnv(2, 0)
+    assert len(dataset) == 50
+    dataset_iter = iter(dataset)
+    assert len(dataset_iter) == 50
+    process_1_1 = list(dataset_iter)
+    assert len(process_1_1) == 50
+    dataset_iter = iter(dataset)
+    assert len(dataset_iter) == 50
+    process_1_2 = list(dataset_iter)
+    assert len(process_1_2) == 50
+
+    dataset = StreamingDataset(name="choco", cache_dir=tmpdir, shuffle=False)
+    dataset.distributed_env = _DistributedEnv(2, 1)
+    assert len(dataset) == 50
+    dataset_iter = iter(dataset)
+    process_2_1 = list(dataset_iter)
+    assert len(process_2_1) == 50
+    dataset_iter = iter(dataset)
+    assert len(dataset_iter) == 50
+    process_2_2 = list(dataset_iter)
+    assert len(process_2_2) == 50
+
+    _, intervals_per_process = dataset.shuffle.get_chunks_and_intervals_per_process(
+        dataset.distributed_env, dataset.current_epoch
+    )
+
+    assert process_1_1 == process_1_2
+
+    found_list = []
+    for i in process_1_1:
+        found = False
+        for interval in intervals_per_process[0]:
+            if interval[0] <= i <= interval[1]:
+                found = True
+                break
+        found_list.append(found)
+
+    assert all(found_list) is True
+
+    found_list = []
+    for i in process_2_1:
+        found = False
+        for interval in intervals_per_process[1]:
+            if interval[0] <= i <= interval[1]:
+                found = True
+                break
+        found_list.append(found)
+
+    assert all(found_list) is True
+
+    assert len([i for i in process_1_1 if i in process_2_1]) == 0
+    assert len([i for i in process_1_2 if i in process_2_2]) == 0
+
+
+def test_streaming_dataset_distributed_across_chunks_shuffle(tmpdir):
```

Comment:
same comment as before, the test doesn't test the actual shuffling strategy

Reply:
add some extra bits.